[DOCS-1682] Document GraphQL naming constraints for metrics (#1481)
wandb/docs: content/en/support/kb-articles/metric-naming-graphql-limitations.md

---
url: /support/:filename

> title: "Why can't I sort or filter metrics with special characters?"

Author: ```suggestion
title: "Why can't I sort or filter metrics with certain characters?"
```

toc_hide: true
type: docs
support:
   - experiments
---

> Due to GraphQL limitations, metrics containing special characters (such as commas, hyphens, spaces, or symbols) may not be sortable or filterable in the W&B UI.

Author: Rephrase this content to be more similar to the copy near https://github.com/wandb/docs/pull/1481/files#diff-f66283c952d7d8e9198480a2d8632ac93ecacdbcce51ffb410501734d21ba80cR71, to explain what characters a metric is allowed to have. Then you can describe the problems you can run into otherwise.

## Valid metric names

> Metric names must follow these rules:

Author: It looks like we can probably create an include with the metric naming rules, and use `readfile` to include that content in multiple places. By convention, includes are in `content/en/_includes/`.

- **Allowed characters**: Letters (A-Z, a-z), digits (0-9), and underscores (_)
- **Starting character**: Must start with a letter or underscore
- **Pattern**: Should match `/^[_a-zA-Z][_a-zA-Z0-9]*$/`

## Examples

> **Valid metric names:**

Author: Similarly, this could be reused, after addressing the feedback I left about it in the log/_index.md.

```python
wandb.log({"accuracy": 0.9, "val_loss": 0.1})
wandb.log({"modelAccuracy": 0.95, "learning_rate": 0.001})
```

**Invalid metric names that may cause sorting issues:**
```python
wandb.log({"acc,val": 0.9})  # Contains comma
wandb.log({"loss-train": 0.1})  # Contains hyphen
wandb.log({"test acc": 0.95})  # Contains space
```

## Recommended solution

> Replace special characters with underscores when naming metrics:

Author: ```suggestion
Replace invalid characters with valid characters such as underscores:
```
